frontend/components/utils/BidInput.vue

```diff
@@ -0,0 +1,149 @@
+import BigNumber from 'bignumber.js';
+import { Tooltip } from 'ant-design-vue';
+import FormatCurrency from './FormatCurrency.vue';
+
+export default Vue.extend({
+    components: { FormatCurrency, Tooltip },
+    props: {
+        totalPrice: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        minimumDepositDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: null,
+        },
+        value: {
+            type: Object as Vue.PropType<BigNumber> | undefined,
+            default: undefined,
+        },
+        disabled: {
+            type: Boolean,
+            default: false,
+        },
+    },
+    data() {
+        return {
+            userInput: undefined as string | undefined,
+            error: undefined as string | undefined,
+        };
+    },
+    computed: {
+        showErrorPopup(): boolean {
+            return !!this.error;
+        },
+        maximumBid(): BigNumber {
+            return this.totalPrice.minus(this.minimumDepositDai);
+        },
+    },
+    watch: {
+        value(newVal: BigNumber | undefined): void {
+            if (!newVal) {
+                this.resetToTotal();
+                return;
+            }
+            if (newVal.isEqualTo(this.totalPrice)) {
+                this.resetToTotal();
+            }
+            if (newVal.isEqualTo(this.minimumDepositDai)) {
+                this.userInput = newVal.toString();
+            }
+        },
+        error(newVal): void {
+            this.$emit('error', newVal);
+        },
+        userInput(newVal, oldVal) {
+            this.validateInput(newVal, oldVal);
+        },
+        totalPrice(newVal: BigNumber | undefined) {
+            if (newVal && newVal.minus(this.minimumDepositDai).isLessThan(this.value)) {
+                this.error = `The bidding amount can not be greater than ${this.maximumBid.toFixed(2)} DAI`;
+            }
+        },
+    },
+    methods: {
+        validateInput(newVal: string, oldVal: string): void {
+            this.error = undefined;
+            if (!newVal) {
+                return;
+            }
+            const value = new BigNumber(newVal);
+            if (isNaN(value.toNumber())) {
+                this.userInput = oldVal;
+            } else if (value.isGreaterThan(this.maximumBid)) {
+                this.error = `The bidding amount can not be greater than ${this.maximumBid.toFixed(2)} DAI`;
+            } else if (value.isLessThan(this.minimumDepositDai)) {
+                this.error = `The bidding amount can not be smaller than ${this.minimumDepositDai.toFixed(2)} DAI`;
+            } else {
+                this.setValue(value);
+            }
+        },
+        hideTotalPrice(): void {
+            if (!this.value) {
+                this.setValue(new BigNumber(0));
+            }
+        },
+        validateFinished() {
+            if (!this.userInput) {
+                this.userInput = '0';
+            }
```

Comment:
Currently, focusing the input and then loosing the focus will result in 0 being set to the input. I would expect that it should return back to total